pulser-core/pulser/sequence/sequence.py

```diff
@@ -193,15 +211,15 @@ def available_channels(self) -> dict[str, Channel]:
         # Show all channels if none are declared, otherwise filter depending
         # on whether the sequence is working on XY mode
         # If already in XY mode, filter right away
-        if not self._channels and not self._in_xy:
+        if not self._schedule and not self._in_xy:
```

Comment:
I agree that sometimes it makes sense to be explicit, but keeping in line with [PEP8 programming recommendations](https://peps.python.org/pep-0008/#programming-recommendations), I prefer to make these checks in this way. 
Is there a particular justification behind your suggestion?

Reply:
I guess PEP8 recommends this idiom for shorter statements, and probably also duck-typing?
My personal insight on this is that using an explicit condition, it could protect the code better, and help to avoid bugs. For example, if `_schedule` happened to become optional in the future, mypy would complain and force the developer to check implications in this function:

- using `len`: the function invocation would fail  on the optional
- using the `== {}` comparison: `self._schedule.values()` would still be optional and fail line 222

I understand this is controversial however (and not wholly matching the Pythonic mentality).

Reply:
I think it's a trade-off between style and robustness. In general, I would prefer not to deviate from PEP8 guidelines, as I feel they make for cleaner (and yes, more pythonic) code. I'm still open to deviations when there is a good reason for it, but in the case of `self._schedule` I feel like it is used extensively I'm sure `mypy` would complain about other parts of the code.

In fact, if we consider your example where `self._schedule` becomes optional, I would argue that we would want the same behaviour to hold on this line (ie self._schedule being empty or not defined at all would both mean that you can choose any channel). 

So, do you think we can agree to disagree on this one? 🙏